compiler/plugins/input/Torch/torch-iree/PluginRegistration.cpp

```diff
@@ -74,6 +83,7 @@ struct TorchSession
   void populateCustomInputConversionTypes(StringSet<> &typeMnemonics) override {
     typeMnemonics.insert("tm_tensor");
     typeMnemonics.insert("torch");
+    typeMnemonics.insert("onnx");
```

Comment:
The `populateDetectedCustomInputConversionTypes` below could also be updated to auto-detect onnx. Is there a particular MLIR dialect that could look for?

Reply:
> Is there a particular MLIR dialect that could look for?

Looks like checking for the existence of a `torch.onnx_meta.opset_version` attr works.

https://github.com/llvm/torch-mlir/blob/b3a3ad4e2a5a6c2a5c649d21cb3568e238cca132/lib/Conversion/TorchOnnxToTorch/TorchOnnxToTorch.cpp#L39-L53

I'll send a PR that teaches the auto input pipeline about onnx. (https://github.com/ScottTodd/iree/tree/onnx-auto, just need tests ... next week). 

---

We should also add a "did conversion succeed" pass to torch. This sort of error comes too late and without enough context:
`D:\dev\projects\iree-tmp\2024_01_19\onnx_leaky_relu_imported.mlir:2:3: error: failed to legalize unresolved materialization from '!torch.vtensor<[3,2,5],f32>' to '!vm.ref<!torch.vtensor<[3,2,5],f32>>' that remained live after conversion`
(see https://github.com/openxla/iree/pull/16033 for related discussions)

Reply:
Sent https://github.com/openxla/iree/pull/16178 for auto detection